hibernate-cache/src/test/java/org/infinispan/test/hibernate/cache/functional/AbstractFunctionalTest.java

```diff
@@ -239,4 +253,114 @@ protected CountDownLatch expectPutWithValue(AdvancedCache cache, Predicate<Objec
 			return new CountDownLatch(0);
 		}
 	}
+
+   protected CountDownLatch expectAfterEndInvalidation(AdvancedCache cache, int numInvalidates) {
+      CountDownLatch latch = new CountDownLatch(numInvalidates);
+      wrapInboundInvocationHandler(cache, handler -> new ExpectingInboundInvocationHandler(handler, latch));
```

Comment:
You don't ever remove that handler. I suggest adding a `List<Runnable> cleanup` collection as a field and executing tasks from that from `@AfterMethod`. `expectAfterEndInvalidation` should add removal of the wrapper to that collection.